src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -2342,6 +2342,65 @@ class CreateStatementSegment(BaseSegment):
         ),
         Ref("IfNotExistsGrammar", optional=True),
         Ref("ObjectReferenceSegment"),
+        # Next set are Storage Integration statements
+        # https://docs.snowflake.com/en/sql-reference/sql/create-storage-integration.html
+        Sequence(
+            AnySetOf(
+                Sequence("TYPE", Ref("EqualsSegment"), "EXTERNAL_STAGE"),
+                Sequence("ENABLED", Ref("EqualsSegment"), Ref("BooleanLiteralGrammar")),
+                OneOf(
+                    Ref("S3StorageIntegrationParameters"),
+                    Ref("GCSStorageIntegrationParameters"),
+                    Ref("AzureStorageIntegrationParameters"),
+                ),
+                # Note Could result in mismatched IntegrationParams and AllowedLocations
+                # E.G. S3 IntegrationParameters would parse with GCS compliant paths
+                # Alternative would be to add these to IntegrationParameter Segments but
+                # I feel like that is harder to understand the match_grammar
+                # when comparing against Snowflake docs.
+                Sequence(
+                    "STORAGE_ALLOWED_LOCATIONS",
+                    Ref("EqualsSegment"),
+                    Bracketed(
+                        Delimited(
+                            OneOf(
+                                Ref("S3Path"),
+                                Ref("GCSPath"),
+                                Ref("AzureBlobStoragePath"),
+                                Ref(
+                                    "QuotedLiteralSegment"
+                                ),  # TODO: Need help. Should only be '*'.
```

Comment:
As this one is used elsewhere, you should add a `QuotedStarSegment` in the section starting at line 89.

```py
    QuotedStarSegment=StringParser(
        "'*'",
        CodeSegment,
        name="quoted_star",
        type="identifier",
        trim_chars=("'",),
    ),
```

Then can change to this:

```suggestion
                                Ref("QuotedStarSegment"),
```